internal/expr/bindtypes.go

```diff
@@ -150,8 +178,80 @@ func (e *columnsInsertExpr) String() string {
 	return fmt.Sprintf("ColumnInsert[%v %v]", e.columns, e.sources)
 }
 
-func (e *columnsInsertExpr) bindTypes(argInfo typeinfo.ArgInfo) (any, error) {
-	return nil, fmt.Errorf("insert input expression not implemented")
+// bindTypes generates a *typedInputExpr containing type information about the
+// columnsInsertExpr. It checks that all the listed columns are provided by the
+// supplied types. If a map with an asterisk is passed, the spare columns are
+// taken from that map.
+func (e *columnsInsertExpr) bindTypes(argInfo typeinfo.ArgInfo) (tie any, err error) {
+	defer func() {
+		if err != nil {
+			err = fmt.Errorf("input expression: %s: %s", err, e.raw)
+		}
+	}()
+
+	// 1. We go over all the supplied types and collect all the columns that
+	// they provide.
+	var remainingMap *string
```

Comment:
Could maybe have a seperate comment about the remaining map.

Reply:
Agree